packages/lib/shared/layouts/PoolLayout.tsx

```diff
@@ -0,0 +1,80 @@
+import { PoolVariant } from '@repo/lib/modules/pool/pool.types'
+import { ChainSlug, getChainSlug, getPoolTypeLabel } from '@repo/lib/modules/pool/pool.utils'
+import { PropsWithChildren, Suspense } from 'react'
+import { PoolDetailSkeleton } from '@repo/lib/modules/pool/PoolDetail/PoolDetailSkeleton'
+import { getApolloServerClient } from '@repo/lib/shared/services/api/apollo-server.client'
+import { GetPoolDocument } from '@repo/lib/shared/services/api/generated/graphql'
+import { Metadata } from 'next'
+import { PoolProvider } from '@repo/lib/modules/pool/PoolProvider'
+import { arrayToSentence } from '@repo/lib/shared/utils/strings'
+import { ensureError } from '@repo/lib/shared/utils/errors'
+import { notFound } from 'next/navigation'
+import { getUserReferenceTokens } from '@repo/lib/modules/pool/pool-tokens.utils'
+
+export type Props = PropsWithChildren<{
+  chain: ChainSlug
+  id: string
+  variant?: PoolVariant
+}>
+
+async function getPoolQuery(chain: ChainSlug, id: string) {
+  const _chain = getChainSlug(chain)
+  const variables = { id: id.toLowerCase(), chain: _chain }
+
+  try {
+    const result = await getApolloServerClient().query({
+      query: GetPoolDocument,
+      variables,
+      context: {
+        fetchOptions: {
+          next: { revalidate: 30 },
+        },
+      },
+    })
+    return { data: result.data, error: null }
+  } catch (error: unknown) {
+    return { data: null, error: ensureError(error) }
+  }
+}
+
+export async function generateMetadata({ id, chain, variant }: Props): Promise<Metadata> {
+  const { data } = await getPoolQuery(chain, id)
+
+  const pool = data?.pool
+  if (!pool) return {}
+
+  const displayTokens = getUserReferenceTokens(pool)
+
+  const poolTokenString = arrayToSentence(displayTokens.map(token => token.symbol))
+
+  return {
+    title: `Liquidity Pool (${variant}): ${pool.name}`,
+    description: `${pool.symbol} is a Balancer ${variant} ${getPoolTypeLabel(
```

Comment:
@groninge01 We were already using Balancer in the previous code. Should we change this description for beets?

Reply:
Using Balancer is fine. But i just took out `${pool.symbol}` with `This` :)